controllers/tasksRequests.js

```diff
@@ -0,0 +1,118 @@
+const { INTERNAL_SERVER_ERROR, SOMETHING_WENT_WRONG } = require("../constants/errorMessages");
+const taskRequestsModel = require("../models/taskRequests");
+const userModel = require("../models/users.js");
+const userStatusModel = require("../models/userStatus.js");
+const { userState } = require("../constants/userStatus");
+const tasksModel = require("../models/tasks.js");
+
+const fetchTaskRequests = async (_, res) => {
+  try {
+    const data = await taskRequestsModel.fetchTaskRequests();
+
+    return res.status(200).json({
+      message: "Task requests returned successfully",
+      data,
+    });
+  } catch (err) {
+    logger.error("Error while fetching task requests", err);
+    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+};
+
+const addOrUpdate = async (req, res) => {
+  try {
+    const { taskId, userId } = req.body;
+    if (!taskId) {
+      return res.boom.badRequest("taskId not provided");
+    }
+    if (!userId) {
+      return res.boom.badRequest("userId not provided");
+    }
+
+    const { userExists } = await userModel.fetchUser({ userId });
+    if (!userExists) {
+      return res.boom.conflict("User does not exist");
+    }
+
+    const { userStatusExists, data: userStatus } = await userStatusModel.getUserStatus(userId);
+    if (!userStatusExists) {
+      return res.boom.conflict("User status does not exist");
```

Comment:
instead of doing 3 such rigorous if checks, how about we create a separate validator function?